src/Stratis.Bitcoin.Features.BlockStore/BlockStoreSignaled.cs

```diff
@@ -104,17 +110,56 @@ protected override void OnNextCore(Block block)
             this.logger.LogTrace("(-)");
         }
 
+        /// <summary>Initializes the <see cref="BlockStoreSignaled"/>.</summary>
+        public void Initialize()
+        {
+            this.localBatch = new List<ChainedBlock>();
+
+            // Configure batch timer.
+            this.batchTimer = new Timer(this.batchInterval.TotalMilliseconds);
+            this.batchTimer.AutoReset = false;
+            this.batchTimer.Elapsed += async (sender, args) =>
+            {
+                using (await this.asyncLock.LockAsync(this.nodeLifetime.ApplicationStopping).ConfigureAwait(false))
+                {
+                    if (!this.batchTimer.Enabled)
+                        return;
+
+                    await SendBatchAsync();
+                }
+            };
+
+            this.blocksToAnnounce = new AsyncQueue<ChainedBlock>(async (item, cancellation) =>
+            {
+                using (await this.asyncLock.LockAsync(cancellation).ConfigureAwait(false))
+                {
+                    this.localBatch.Add(item);
+
+                    // Send batch right away if tip, if not and the timer haven't been started already- start the timer.
+                    if (item == this.chain.Tip)
+                    {
+                        this.batchTimer.Stop();
+
+                        await SendBatchAsync();
+                    }
+                    else if (!this.batchTimer.Enabled)
+                    {
+                        this.batchTimer.Start();
+                    }
+                }
+            });
+        }
+
         /// <summary>
-        /// A loop method that continuously relays blocks found in <see cref="blocksToAnnounce"/> to connected peers on the network.
+        /// A method that relays blocks found in <see cref="localBatch"/> to connected peers on the network.
         /// </summary>
         /// <remarks>
         /// <para>
-        /// The queue <see cref="blocksToAnnounce"/> contains
-        /// hashes of blocks that were validated by the consensus rules.
+        /// The list <see cref="localBatch"/> contains hashes of blocks that were validated by the consensus rules.
         /// </para>
         /// <para>
         /// This block hashes need to be relayed to connected peers. A peer that does not have a block
-        /// will then ask for the entire block, that means only blocks that have been stored should be relayed.
+        /// will then ask for the entire block, that means only blocks that have been stored\cached should be relayed.
```

Comment:
maybe use / instead of \ just to prevent problems

Reply:
Fixed.